# Problema 1

Considere la sucesión $a_{n} = 3n^{2} + 7$. Por medio de un código en *Python* despliega en pantalla:

1. Los primeros $5$ términos.
2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [1]:
a=[]
for i in range(25):
    n=i+1
    a.append(3*n**2+7)
    print("Término",n,"=",a[i])

Término 1 = 10
Término 2 = 19
Término 3 = 34
Término 4 = 55
Término 5 = 82
Término 6 = 115
Término 7 = 154
Término 8 = 199
Término 9 = 250
Término 10 = 307
Término 11 = 370
Término 12 = 439
Término 13 = 514
Término 14 = 595
Término 15 = 682
Término 16 = 775
Término 17 = 874
Término 18 = 979
Término 19 = 1090
Término 20 = 1207
Término 21 = 1330
Término 22 = 1459
Término 23 = 1594
Término 24 = 1735
Término 25 = 1882


1. Los primeros $5$ términos.

In [2]:
for i in range (5):
    n=i+1
    print("Término",n,"=",a[i])

Término 1 = 10
Término 2 = 19
Término 3 = 34
Término 4 = 55
Término 5 = 82


2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

In [3]:
for i in range (15,20,1):
    n=i+1
    print("Término",n,"=",a[i])

Término 16 = 775
Término 17 = 874
Término 18 = 979
Término 19 = 1090
Término 20 = 1207


# Problema 2

Considere la sucesión $g_{n} = 5n^{3}$. Implemente un código en *Python* que muestre en pantalla:

1. Los primeros $4$ términos de la sucesión.
2. Los $4$ términos que vienen inmediatamente después del octavo término.
3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [1]:
g=[]
for i in range(30):
    n=i+1
    g.append(5*n**3)
    print("Término",n,":",g[i])


Término 1 : 5
Término 2 : 40
Término 3 : 135
Término 4 : 320
Término 5 : 625
Término 6 : 1080
Término 7 : 1715
Término 8 : 2560
Término 9 : 3645
Término 10 : 5000
Término 11 : 6655
Término 12 : 8640
Término 13 : 10985
Término 14 : 13720
Término 15 : 16875
Término 16 : 20480
Término 17 : 24565
Término 18 : 29160
Término 19 : 34295
Término 20 : 40000
Término 21 : 46305
Término 22 : 53240
Término 23 : 60835
Término 24 : 69120
Término 25 : 78125
Término 26 : 87880
Término 27 : 98415
Término 28 : 109760
Término 29 : 121945
Término 30 : 135000


1. Los primeros $4$ términos de la sucesión.

In [2]:
for i in range(4):
    n=i+1
    print("Término",n,":",g[i])

Término 1 : 5
Término 2 : 40
Término 3 : 135
Término 4 : 320


2. Los $4$ términos que vienen inmediatamente después del octavo término.


In [3]:
for i in range(8,12):
    n=i+1
    print("Término",n,":",g[i])

Término 9 : 3645
Término 10 : 5000
Término 11 : 6655
Término 12 : 8640


3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

In [9]:
final = 40000
termino = 0
n = 0
dif = final-termino
while dif > 0:
    n += 1
    termino = g[n-1]
    dif = final-termino

if dif==0:
    print("El término",final,"está en la posición",n)
else:
    print("El término",final,"no está en la sucesión")


El término 40000 está en la posición 20


Veamos con 23000:

In [10]:
final = 23000
termino = 0
n = 0
dif = final-termino
while dif > 0:
    n += 1
    termino = g[n-1]
    dif = final-termino

if dif==0:
    print("El término",final,"está en la posición",n)
else:
    print("El término",final,"no está en la sucesión")

El término 23000 no está en la sucesión


# Problema 3

La sucesión de Fibonacci es una serie de números en la que cada número, desde el tercero en adelante, es la suma de los dos anteriores. Comienza con el $0$ y $1$, tal como se observa a continuación:

$$0, 1, 1, 2, 3, 5, 8, 13, ...$$

Formalmente, la sucesión se define de la siguiente manera:

\begin{split}
F(0) &= 0\\
F(1) &= 1\\
F(n) &= F(n - 1) + F(n - 2), \ \text{ para $n \geq 2$}
\end{split}

Esta sucesión se utiliza en una variedad de contextos en informática y algoritmos debido a sus propiedades matemáticas y patrones.

1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

In [4]:
F=[0,1]
for i in range(2,30):
    n=i+1
    F.append(F[i-1]+F[i-2])

for i in range(30):
    n=i+1
    print("Término",n,":",F[i])


Término 1 : 0
Término 2 : 1
Término 3 : 1
Término 4 : 2
Término 5 : 3
Término 6 : 5
Término 7 : 8
Término 8 : 13
Término 9 : 21
Término 10 : 34
Término 11 : 55
Término 12 : 89
Término 13 : 144
Término 14 : 233
Término 15 : 377
Término 16 : 610
Término 17 : 987
Término 18 : 1597
Término 19 : 2584
Término 20 : 4181
Término 21 : 6765
Término 22 : 10946
Término 23 : 17711
Término 24 : 28657
Término 25 : 46368
Término 26 : 75025
Término 27 : 121393
Término 28 : 196418
Término 29 : 317811
Término 30 : 514229


1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.


In [5]:
for i in range(20):
    n=i+1
    print("Término",n,":",F[i])

Término 1 : 0
Término 2 : 1
Término 3 : 1
Término 4 : 2
Término 5 : 3
Término 6 : 5
Término 7 : 8
Término 8 : 13
Término 9 : 21
Término 10 : 34
Término 11 : 55
Término 12 : 89
Término 13 : 144
Término 14 : 233
Término 15 : 377
Término 16 : 610
Término 17 : 987
Término 18 : 1597
Término 19 : 2584
Término 20 : 4181


2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

In [6]:
print(sum(F))

1346268


In [7]:
suma=0
for i in range(30):
    suma += F[i]

print(suma)

1346268


# Problema 4

Implementa un código que permita guardar en una lista la sucesión de los primeros $n$ números pares, con $n \geq 1$. Luego, determina:

1. ¿Cuál es el valor del décimo término?
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.
3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

In [11]:
p=[]
for i in range(100):
    n=i+1
    p.append(2*n)
    print("Término",n,":",p[i])

Término 1 : 2
Término 2 : 4
Término 3 : 6
Término 4 : 8
Término 5 : 10
Término 6 : 12
Término 7 : 14
Término 8 : 16
Término 9 : 18
Término 10 : 20
Término 11 : 22
Término 12 : 24
Término 13 : 26
Término 14 : 28
Término 15 : 30
Término 16 : 32
Término 17 : 34
Término 18 : 36
Término 19 : 38
Término 20 : 40
Término 21 : 42
Término 22 : 44
Término 23 : 46
Término 24 : 48
Término 25 : 50
Término 26 : 52
Término 27 : 54
Término 28 : 56
Término 29 : 58
Término 30 : 60
Término 31 : 62
Término 32 : 64
Término 33 : 66
Término 34 : 68
Término 35 : 70
Término 36 : 72
Término 37 : 74
Término 38 : 76
Término 39 : 78
Término 40 : 80
Término 41 : 82
Término 42 : 84
Término 43 : 86
Término 44 : 88
Término 45 : 90
Término 46 : 92
Término 47 : 94
Término 48 : 96
Término 49 : 98
Término 50 : 100
Término 51 : 102
Término 52 : 104
Término 53 : 106
Término 54 : 108
Término 55 : 110
Término 56 : 112
Término 57 : 114
Término 58 : 116
Término 59 : 118
Término 60 : 120
Término 61 : 122
Término 62 : 124
Término 

1. ¿Cuál es el valor del décimo término?

In [12]:
print("El décimo número par es",p[9])

El décimo número par es 20


2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.


In [13]:
print(sum(p))

10100


In [14]:
suma=0
for i in range(100):
    suma += p[i]

print(suma)

10100


3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

In [15]:
final = 58
termino = 0
n = 0
dif = final-termino
while dif > 0:
    n += 1
    termino = p[n-1]
    dif = final-termino

if dif==0:
    print("El término",final,"está en la posición",n)
else:
    print("El término",final,"no está en la sucesión")

El término 58 está en la posición 29


In [16]:
final = 73
termino = 0
n = 0
dif = final-termino
while dif > 0:
    n += 1
    termino = p[n-1]
    dif = final-termino

if dif==0:
    print("El término",final,"está en la posición",n)
else:
    print("El término",final,"no está en la sucesión")

El término 73 no está en la sucesión


# Problema 5

Un nuevo software tiene $500$ usuarios y se espera que el número de usuarios crezca un $15\%$ respecto al mes anterior. Considere que $b_{n}$ es el número de usuarios transcurridos $n$ meses.

1. Calcula cuántos usuarios tendrá el software al cabo de tres meses. ¿Y al cabo de seis meses?
2. Mediante un código en *Pyhton* muestra la cantidad de usuarios que tendrá el software cada mes durante el primer año.
2. ¿Cuántos usuarios tendrá en total el software luego de un año? Utilice ciclos *for* para calcular.

# Problema 6

Un estudiante de ingeniería informática planea ahorrar para un proyecto personal. Para esto, el primer mes deposita en una cuenta $\$12.000$, el segundo mes deposita $\$14.000$ y así sucesivamente, de modo que cada mes aumenta el depósito en $\$2.000$. Considere que $a_{i}$ corresponde al depósito del estudiante en el mes $i$.

1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.
2. ¿Cuánto depositó el estudiante en febrero del segundo año?
3. ¿Cuál será el total ahorrado al finalizar dos años?

In [8]:
a1=12000
d=2000
a=[]
for i in range(24):
    n=i+1
    a.append(a1+(n-1)*d)
    print("El mes",n,"ahorró $",a[i])
    

El mes 1 ahorró $ 12000
El mes 2 ahorró $ 14000
El mes 3 ahorró $ 16000
El mes 4 ahorró $ 18000
El mes 5 ahorró $ 20000
El mes 6 ahorró $ 22000
El mes 7 ahorró $ 24000
El mes 8 ahorró $ 26000
El mes 9 ahorró $ 28000
El mes 10 ahorró $ 30000
El mes 11 ahorró $ 32000
El mes 12 ahorró $ 34000
El mes 13 ahorró $ 36000
El mes 14 ahorró $ 38000
El mes 15 ahorró $ 40000
El mes 16 ahorró $ 42000
El mes 17 ahorró $ 44000
El mes 18 ahorró $ 46000
El mes 19 ahorró $ 48000
El mes 20 ahorró $ 50000
El mes 21 ahorró $ 52000
El mes 22 ahorró $ 54000
El mes 23 ahorró $ 56000
El mes 24 ahorró $ 58000


1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.


In [9]:
for i in range(12):
    print("El mes",i+1,"ahorró $",a[i])

El mes 1 ahorró $ 12000
El mes 2 ahorró $ 14000
El mes 3 ahorró $ 16000
El mes 4 ahorró $ 18000
El mes 5 ahorró $ 20000
El mes 6 ahorró $ 22000
El mes 7 ahorró $ 24000
El mes 8 ahorró $ 26000
El mes 9 ahorró $ 28000
El mes 10 ahorró $ 30000
El mes 11 ahorró $ 32000
El mes 12 ahorró $ 34000


2. ¿Cuánto depositó el estudiante en febrero del segundo año?


In [10]:
b=14
print("El mes",b,"ahorró $",a[b-1])


El mes 14 ahorró $ 38000


3. ¿Cuál será el total ahorrado al finalizar dos años?

In [11]:
print(sum(a))

840000


In [13]:
suma=0
for i in range(24):
    suma += a[i]

print("En 2 años ha ahorrado $",suma)

En 2 años ha ahorrado $ 840000


# Problema 7

Por lo general, los proyectos son organizados en varias etapas que deben completarse secuencialmente.

La fase inicial de un proyecto de software tiene una duración de $3$ meses y cada fase subsecuente reduce su duración en un $10 \%$ debido a la experiencia ganada. Determine:

1. ¿Cuánto dura la quinta fase?
2. Calcula la duración total del proyecto después de $10$ fases.

# Problema 8

En una sucesión aritmética, la diferencia entre un término y el anterior es $4$ y el décimo término es $-20$. Determine:

1. El primer término de la sucesión.
2. La expresión del término de lugar $n$.
3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.
4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

1. El primer término de la sucesión.


Tenemos:
$$a1=x$$
$$d=4$$
Calculamos con la ecuación: 
$$a_{n}=a_{1}+(n-1)d$$
$$-20=x+(10-1)4$$
$$x=-56$$

El primer término es -56.

2. La expresión del término de lugar $n$.


$$a_{n}=a_{1}+(n-1)d$$

$$a_{n}=-56+(n-1)4$$

3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.


In [1]:
a=100
print("El término en el lugar",a,"es",-56+(a-1)*4)

El término en el lugar 100 es 340


Generamos la sucesión "completa":

In [2]:
a1=-56
d=4
a=[]
for i in range(500):
    n=i+1
    a.append(a1+(n-1)*d)
    print("Término",n,":",a[i])

Término 1 : -56
Término 2 : -52
Término 3 : -48
Término 4 : -44
Término 5 : -40
Término 6 : -36
Término 7 : -32
Término 8 : -28
Término 9 : -24
Término 10 : -20
Término 11 : -16
Término 12 : -12
Término 13 : -8
Término 14 : -4
Término 15 : 0
Término 16 : 4
Término 17 : 8
Término 18 : 12
Término 19 : 16
Término 20 : 20
Término 21 : 24
Término 22 : 28
Término 23 : 32
Término 24 : 36
Término 25 : 40
Término 26 : 44
Término 27 : 48
Término 28 : 52
Término 29 : 56
Término 30 : 60
Término 31 : 64
Término 32 : 68
Término 33 : 72
Término 34 : 76
Término 35 : 80
Término 36 : 84
Término 37 : 88
Término 38 : 92
Término 39 : 96
Término 40 : 100
Término 41 : 104
Término 42 : 108
Término 43 : 112
Término 44 : 116
Término 45 : 120
Término 46 : 124
Término 47 : 128
Término 48 : 132
Término 49 : 136
Término 50 : 140
Término 51 : 144
Término 52 : 148
Término 53 : 152
Término 54 : 156
Término 55 : 160
Término 56 : 164
Término 57 : 168
Término 58 : 172
Término 59 : 176
Término 60 : 180
Término 61 : 184
Té

In [3]:
print("El término en el lugar 100 es",a[99])

El término en el lugar 100 es 340


4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

In [4]:
final = 1680
termino = 0
n = 0
dif = final-termino
while dif > 0:
    n += 1
    termino = a[n-1]
    dif = final-termino

if dif==0:
    print("El término",final,"está en la posición",n)
else:
    print("El término",final,"no está en la sucesión")

El término 1680 está en la posición 435


# Problema 9

El tercer término de una sucesión geométrica es $5$ y el sexto término es $40$. Determine:

1. La razón geométrica de la sucesión.
2. El primer término de la sucesión.
3. La expresión del término de lugar $n$.
4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.
4. ¿Qué lugar ocupa en esta sucesión el término $20.480$? Calcule utilizando un ciclo *while*.

# Problema 10

Una empresa de tecnología está optimizando sus servidores. Para ello mide las transacciones por segundo (TPS) de sus servidores en cada mes. Se ha obtenido que en el segundo mes, el rendimiento del servidor fue $1,2$ miles de TPS y en el quinto mes, el rendimiento fue $2,7$ miles de TPS. Si el número de TPS que hay mes a mes sigue una sucesión geométrica, determina utilizando un código en *Python*:

1. ¿Cuál es el rendimiento el noveno mes?
2. ¿En qué mes el rendimiento es de $23,4$ miles de TPS?

Utilice dos decimales para el cálculo.

# Problema 11

Una aplicación de redes sociales se estrenó el $1$ de enero con una cierta cantidad de usuarios. A partir de ahí, el número de usuarios nuevos de cada día fue siempre el mismo. El $3$ de enero tenía $500$ usuarios en total y el $10$ de enero tenía $1.501$ usuarios totales.Determine:

1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?
2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?
3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?



# Problema 12

Una empresa de publicidad ofrece para un puesto de trabajo un salario inicial y un aumento mensual en una razón constante. Javier se presenta a la entrevista, y el encargado del departamento de recursos humanos le explica que en el segundo mes de trabajo su sueldo será de $\$402.000$, y que en el tercer mes de trabajo será de $\$404.010$. Determine:

1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.
2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

# Problema 13

Una empresa de ciberseguridad capacita trabajadores cada trimestre. En el primer trimestre capacitaron a $30$ trabajadores, en el segundo trimestre a $60$ trabajadores, y en el tercer trimestre a $120$ trabajadores. Si en los siguientes trimestres la empresa mantiene el mismo patrón de crecimiento, determine utilizando *Python*:

1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?
2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

# Problema 14

Al llenar un tambor con agua, se registra que durante el minuto $4$ el tambor recibe $128$ litros, y durante el minuto $9$ recibe $88$ litros. Sabiendo que en cada minuto recibe una cantidad $d$ de litros de agua menos que en el minuto anterior, determine:

1. ¿En qué instante el tambor deja de recibir agua?
2. ¿Cuántos litros recibe en total el tambor?

Realice los cálculos anteriores utilizando *Python*.